src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -0,0 +1,386 @@
+/**
+ *  Copyright (C) 2010-2021 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+let kHTMLATagPattern = "(?i)<a([^>]+)>(.+?)</a>"
+let kHTMLAHrefTagPattern = "href=\"(.*?)\""
+
+@objc extension ProjectDetailStoreViewController {
+
+    static var height: CGFloat = Util.screenHeight()
+
+    func createProjectDetailView(_ project: CatrobatProject, target: Any?) -> UIView? {
+        let view = UIView(frame: CGRect(x: 0, y: 0, width: Util.screenWidth(), height: 0))
+        view.backgroundColor = UIColor.clear
+        view.autoresizingMask = .flexibleHeight
+        if Project.projectExists(withProjectID: project.projectID) {
```

Comment:
These lines have no affect here, since the buttons are not added yet. Please move to the end of this method, after the buttons have been added. See this error by opening an existing project in the community. Compare the results with the store version (no open and "re-download" buttons).